### PR TITLE
Add subcommand `dcos service log` in Go

### PR DESCRIPTION
**Pull request by rgo3**
https://jira.mesosphere.com/browse/DCOS_OSS-4784

### CODE REVIEW

File: pkg/cmd/service/service_log.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+func marathonMasterURL(c *marathon.Client, ctx api.Context) (string, error) {
+	ipsRes := make(chan ipsResult)
+	go publicIPs(ipsRes)
+
+	leader, err := c.API.Leader()
+	if err != nil {
+		return "", err
+	}
+
+	publicIPs := <-ipsRes
```
Comment (bamarni) [inline pkg/cmd/service/service_log.go]: `publicIPs` is shadowing the function, can we have a different name?

File: pkg/cmd/service/service_log.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	marathonIP := strings.Split(leader, ":")[0]
+
+	cluster, err := ctx.Cluster()
+	if err != nil {
+		return "", err
+	}
+	// from <scheme>://<url> get the <scheme> field
+	scheme := strings.Split(cluster.URL(), "://")[0]
+
+	return fmt.Sprintf("%s://%s", scheme, publicIPs.ips[marathonIP][0]), nil
```
Comment (bamarni) [inline pkg/cmd/service/service_log.go]: Can we validate that the IP entry exists? Otherwise it can panic.

File: pkg/cmd/service/service_log.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+	var appIDs []string
+	for _, a := range apps.Apps {
+		if (*a.Labels)["DCOS_PACKAGE_FRAMEWORK_NAME"] == service {
+			appIDs = append(appIDs, a.ID)
+		}
+	}
+
+	switch len(appIDs) {
+	case 0:
+		return "", fmt.Errorf("No marathon apps found for service name [%s]", service)
```
Comment (bamarni) [inline pkg/cmd/service/service_log.go]: errors should start with lowercase

File: pkg/pluginutil/pluginutil.go [hunk 4]
```diff
@@ -45,6 +47,43 @@ func HTTPClient(baseURL string, opts ...httpclient.Option) *httpclient.Client {
 	return httpclient.New(strings.TrimRight(baseURL, "/"), append(baseOpts, opts...)...)
 }
 
+// NewHTTPClient returns an HTTP client from a plugin runtime.
+// It is created with the `client-go` package and will eventually
+// replace the httpClient of the CLI.
+func NewHTTPClient(baseURL string) *http.Client {
```
Comment (bamarni) [inline pkg/pluginutil/pluginutil.go:53]: Just noticed we don't set our CLI specific user agent when we use the httpclient from client-go in general, but I think it requires change in the package first.

File: pkg/cmd/service/service_log.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+				Follow: follow,
+				Format: output,
+				Skip:   -1 * lines,
+			}
+			return logClient.PrintTask(taskID, file, opts)
+		},
+	}
+
+	cmd.Flags().BoolVar(&follow, "follow", false, "Dynamically update the log")
+	cmd.Flags().IntVar(&lines, "lines", 10, "Dynamically update the log")
```
Comment (armandgrillet) [inline pkg/cmd/service/service_log.go]: Wrong description.

File: pkg/cmd/service/service_log.go [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+	cmd.Flags().StringVarP(&output, "output", "o", "short", "Format log message output")
+	cmd.Flags().String("user", "", "The SSH user for Marathon")
+	cmd.Flags().String("ssh-config-file", "", "Path to SSH configuration file. This is depracated.")
```
Comment (armandgrillet) [inline pkg/cmd/service/service_log.go]: s/depracated/deprecated

File: pkg/cmd/service/service_log.go [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+	app, err := c.API.Application(appID)
+	if err != nil {
+		return "", err
+	}
+
+	switch len(app.Tasks) {
+	case 1:
+		return app.Tasks[0].ID , nil
+	default:
+		return "", fmt.Errorf("We expected marathon app [%s] to be running 1 task, but we instead found %d tasks", appID, len(app.Tasks))
```
Comment (armandgrillet) [inline pkg/cmd/service/service_log.go]: We rarely use "We" when logging something, we can remove it.

**Review by armandgrillet — CHANGES_REQUESTED** [review]

File: pkg/cmd/service/service_log.go (hunk 7 above)
Comment (rgo3) [reply pkg/cmd/service/service_log.go]: Already changed it, that was copied from the messages of the old code.

**Review by armandgrillet — APPROVED** [review]

**Pull request merged**